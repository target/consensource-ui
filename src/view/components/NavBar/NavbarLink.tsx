import React from 'react';
import { Typography, makeStyles, createStyles, Theme } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { useSelectedRoute } from 'services/hooks';

const useStyles = makeStyles(({ spacing, palette }: Theme) =>
  createStyles({
    link: {
      textDecoration: 'none',
      color: palette.primary.light,
      marginRight: spacing(4),
      '&:hover': {
        color: palette.primary.contrastText,
      },
    },
    selected: {
      color: palette.primary.contrastText,
    },
  }),
);

export interface NavbarLink {
  label: string;
  route: string;
}

export const NavbarLink = ({ label, route }: NavbarLink) => {
  const classes = useStyles();
  const isSelected = useSelectedRoute(route);

  return (
    <Link to={route} className={classes.link}>
      <Typography
        variant="body1"
        className={`${isSelected && classes.selected}`}
      >
        {label}
      </Typography>
    </Link>
  );
};
