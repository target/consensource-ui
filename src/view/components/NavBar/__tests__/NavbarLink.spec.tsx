import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { NavbarLink } from '../NavbarLink';

describe('<NavbarLink />', () => {
  it('renders an unselected link', () => {
    const label = 'label';

    const { container, getByText } = render(
      <MemoryRouter initialEntries={['/test']}>
        <Route path="/test">
          <NavbarLink route="/" label={label} />
        </Route>
      </MemoryRouter>,
    );

    expect(getByText(label).className.includes('selected')).toBe(false);
    expect(container).toMatchSnapshot();
  });

  it('renders a selected link', () => {
    const label = 'label';

    const { container, getByText } = render(
      <MemoryRouter initialEntries={['/test']}>
        <Route path="/test">
          <NavbarLink route="/test" label={label} />
        </Route>
      </MemoryRouter>,
    );

    expect(getByText(label).className.includes('selected')).toBe(true);
    expect(container).toMatchSnapshot();
  });

  it('sets the link target to "_blank" if "openInNewTab" is true', () => {
    const { getByRole } = render(
      <MemoryRouter initialEntries={['/test']}>
        <Route path="/test">
          <NavbarLink route="/test" label="test" openInNewTab />
        </Route>
      </MemoryRouter>,
    );

    expect(getByRole('link')).toHaveAttribute('target', '_blank');
  });

  it('sets the link target to "" if "openInNewTab" is false', () => {
    const { getByRole } = render(
      <MemoryRouter initialEntries={['/test']}>
        <Route path="/test">
          <NavbarLink route="/test" label="test" />
        </Route>
      </MemoryRouter>,
    );

    expect(getByRole('link')).toHaveAttribute('target', '');
  });
});
