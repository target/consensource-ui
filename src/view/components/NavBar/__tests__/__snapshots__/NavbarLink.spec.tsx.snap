// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarLink /> renders a selected link 1`] = `
<div>
  <a
    class="makeStyles-link-6"
    href="/test"
    target=""
  >
    <p
      class="MuiTypography-root makeStyles-link-4 makeStyles-selected-5 MuiTypography-body1"
    >
      label
    </p>
  </a>
</div>
`;

exports[`<NavbarLink /> renders an unselected link 1`] = `
<div>
  <a
    class="makeStyles-link-3"
    href="/"
    target=""
  >
    <p
      class="MuiTypography-root makeStyles-link-1 false MuiTypography-body1"
    >
      label
    </p>
  </a>
</div>
`;
